### PR TITLE
Updated ApiClient and Services

### PR DESCRIPTION
**Pull request by alesyalis**

### CODE REVIEW

File: Exadel.Compreface.AcceptenceTests/Services/FaceDetectionServiceTest.cs [hunk 1]
```diff
@@ -12,7 +12,7 @@ public class FaceDetectionServiceTest
     {
         private readonly FaceDetectionService _faceDetectionService;
 
-        private readonly FaceDetectionRequest _faceDetectionRequest;
+        private readonly FaceDetectionRequestByFileRath _faceDetectionRequest;
```
Comment (vldmrklnk) [inline Exadel.Compreface.AcceptenceTests/Services/FaceDetectionServiceTest.cs]: Rath=>Path

File: Exadel.Compreface/Helpers/ConvertUrlToBase64StringHelpers.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
+﻿using Exadel.Compreface.Clients.ApiClient;
+using Flurl.Http;
```
Comment (vldmrklnk) [inline Exadel.Compreface/Helpers/ConvertUrlToBase64StringHelpers.cs]: Unnecessary reference should be removed.

File: Exadel.Compreface/Clients/ApiClient/IApiClient.cs [hunk 3]
```diff
@@ -28,5 +28,6 @@ public interface IApiClient
         Task<TResponse> PutJsonAsync<TResponse>(string requestUrl, object body, HttpCompletionOption completionOption = HttpCompletionOption.ResponseContentRead, CancellationToken cancellationToken = default);
 
         Task<TResponse> PutJsonAsync<TResponse>(Url requestUrl, object body, HttpCompletionOption completionOption = HttpCompletionOption.ResponseContentRead, CancellationToken cancellationToken = default);
+        Task<byte[]> GetBytesAsync(string url, HttpCompletionOption completionOption = HttpCompletionOption.ResponseContentRead, CancellationToken cancellationToken = default);
```
Comment (vldmrklnk) [inline Exadel.Compreface/Clients/ApiClient/IApiClient.cs:32]: Between methods should be empty row to improve readability.

File: Exadel.Compreface/Services/FaceDetectionService.cs [hunk 4]
```diff
@@ -22,62 +23,57 @@ public FaceDetectionService(IComprefaceConfiguration configuration)
...
-            response = await
+            var requestUrlWithQueryParameters = GetRequestUrl(faceDetectionRequest);
+            var response = await
                 _apiClient.PostMultipartAsync<FaceDetectionResponse>(
                     requestUrl: requestUrlWithQueryParameters,
                     buildContent: mp =>
                         mp.AddFile("file", fileName: FileHelpers.GenerateFileName(faceDetectionRequest.FilePath), path: faceDetectionRequest.FilePath));
 
             return response;
         }
```
Comment (vldmrklnk) [inline Exadel.Compreface/Services/FaceDetectionService.cs:37]: Between methods should be empty row to improve readability.

File: Exadel.Compreface/Services/FaceVerificationService.cs [hunk 5]
```diff
@@ -91,4 +71,16 @@ public async Task<FaceVerificationResponse> VerifyAsync(FaceVerificationWithBase
 
         return response;
     }
```
Comment (vldmrklnk) [inline Exadel.Compreface/Services/FaceVerificationService.cs:74]: Between methods should be empty row to improve readability.

File: Exadel.Compreface/Services/FaceVerificationService.cs [hunk 6]
```diff
@@ -67,22 +41,28 @@ public async Task<FaceVerificationResponse> VerifyAsync(FaceVerificationRequest
...
```
Comment (vldmrklnk) [inline Exadel.Compreface/Services/FaceVerificationService.cs:43]: Between methods should be empty row to improve readability.

File: Exadel.Compreface/Services/RecognitionService/FaceCollection.cs [hunk 7]
```diff
@@ -151,5 +140,15 @@ public async Task<byte[]> DownloadAsync(DownloadImageBySubjectIdRequest download
 
             return response;
         }
```
Comment (vldmrklnk) [inline Exadel.Compreface/Services/RecognitionService/FaceCollection.cs:144]: Between methods should be empty row to improve readability.

File: Exadel.Compreface/Services/RecognitionService/FaceCollection.cs [hunk 8]
```diff
@@ -26,53 +29,39 @@ public FaceCollection(IComprefaceConfiguration configuration)
...
+                   mp.AddFile("file", fileName: FileHelpers.GenerateFileName(request.FilePath), path: request.FilePath));
 
-                var addBase64SubjectExampleRequest = new AddBase64SubjectExampleRequest()
-                {
-                    DetProbThreShold = request.DetProbThreShold,
-                    File = fileInBase64String,
-                    Subject = request.Subject,
-                };
+            return response;
+        }
```
Comment (vldmrklnk) [inline Exadel.Compreface/Services/RecognitionService/FaceCollection.cs:41]: Between methods should be empty row to improve readability.

File: Exadel.Compreface/Services/RecognitionService/RecognizeFaceFromImage.cs [hunk 9]
```diff
@@ -35,29 +35,36 @@ public async Task<RecognizeFaceFromImageResponse> RecognizeAsync(RecognizeFaceFr
...
-                var fileInBase64String = Convert.ToBase64String(fileStream);
+            var response = await
+                _apiClient.PostMultipartAsync<RecognizeFaceFromImageResponse>(
+                    requestUrl: requestUrlWithQueryParameters,
+                    buildContent: mp =>
+                    mp.AddFile("file", fileName: FileHelpers.GenerateFileName(request.FilePath), path: request.FilePath));
 
-                var addBase64SubjectExampleRequest = new AddBase64SubjectExampleRequest()
+            return response;
+        }
```
Comment (vldmrklnk) [inline Exadel.Compreface/Services/RecognitionService/RecognizeFaceFromImage.cs:45]: Between methods should be empty row to improve readability.

File: Exadel.Compreface/Services/RecognitionService/FaceCollection.cs [hunk 10]
```diff
@@ -9,9 +9,12 @@
 using Exadel.Compreface.DTOs.ExampleSubjectDTOs.DownloadImageBySubjectId;
 using Exadel.Compreface.DTOs.ExampleSubjectDTOs.ListAllExampleSubject;
 using Exadel.Compreface.DTOs.HelperDTOs;
+using Exadel.Compreface.DTOs.HelperDTOs.BaseDTOs;
+using Exadel.Compreface.DTOs.SubjectExampleDTOs.AddSubjectExample;
 using Exadel.Compreface.Helpers;
 using Flurl;
 using Flurl.Http;
+using System;
```
Comment (vldmrklnk) [inline Exadel.Compreface/Services/RecognitionService/FaceCollection.cs]: Seems that this reference is unnecessary.

File: Exadel.Compreface/DTOs/SubjectExampleDTOs/AddBase64SubjectExample/AddBase64SubjectExampleRequest.cs [hunk 11]
```diff
@@ -1,10 +1,12 @@
...
+namespace Exadel.Compreface.DTOs.ExampleSubjectDTOs.AddBase64ExampleSubject
 {
-    public class AddBase64SubjectExampleRequest
+    public class AddBase64SubjectExampleRequest : BaseExampleRequest
     {
-        public string Subject { get; set; }
+        //public string Subject { get; set; }
 
-        public decimal? DetProbThreShold { get; set; }
+        //public decimal? DetProbThreShold { get; set; }
```
Comment (Ukrainiangirl) [inline Exadel.Compreface/DTOs/SubjectExampleDTOs/AddBase64SubjectExample/AddBase64SubjectExampleRequest.cs]: delete it

File: SimpleExample/Program.cs [hunk 12]
```diff
@@ -15,9 +18,34 @@
...
+var configdetect = configuration.GetSection("FaceDetectionApiKey").Value;
+var detectService = client.GetCompreFaceService<FaceDetectionService>(configdetect);
+var result = await detectService.DetectAsync(faceDetectionRequest);
+
+var addSubjectExampleRequest = new AddSubjectExampleRequestByFileUrl()
+{
+    Subject = "API",
+    FileUrl = "http://t1.gstatic.com/licensed-image?q=tbn:ANd9GcS6j6jazxVpBpf4ZdKMgUeN61C_s4EJPqI7NijX0PDep3SGXQ_bT9ap12h2MWoIolUIpngs3pApkIH-Kzw",
+    //FilePath = "https://raw.githubusercontent.com/exadel-inc/compreface-net-sdk/main/Exadel.Compreface.AcceptenceTests/Resources/Images/brad-pitt_24.jpg",
+};
```
Comment (Ukrainiangirl) [inline SimpleExample/Program.cs]: delete it

File: SimpleExample/appsettings.json [hunk 13]
```diff
@@ -3,5 +3,5 @@
   "Port": "8000",
   "FaceDetectionApiKey": "59e2df37-0781-439d-802a-63329bb7e60a",
   "FaceVerificationApiKey": "2283c5cf-4a8b-49d9-9bd7-0a75f49c816b",
-  "FaceRecognitionApiKey": "00000000-0000-0000-0000-000000000002"
+  "FaceRecognitionApiKey": "e3e83f9e-9dee-4bbd-b79e-eaa20ffed167"
```
Comment (Ukrainiangirl) [inline SimpleExample/appsettings.json]: delete Guids

**Review by Ukrainiangirl — APPROVED** [review]

**Pull request merged**